transformer_engine/paddle/layer/base.py

```diff
@@ -280,6 +338,21 @@ def grad_output_preprocess(
 
         return grad_output_mat, grad_output_c, grad_output_t, bgrad
 
+    def set_nccl_overlap_warning_if_tp(self) -> None:
+        """When using TP, the NCCL communication needs to be scheduled
+        before the GEMM for there to be a guaranteed overlap. From the
```

Comment:
When using TP, NCCL communication needs to be scheduled before GEMM to guarantee overlap.